README.md

```diff
@@ -22,6 +22,297 @@ development:
+
+To store spatial data, you must create a column with a spatial type. PostGIS
+provides a variety of spatial types, including point, linestring, polygon, and
+different kinds of collections. These types are defined in a standard produced
+by the Open Geospatial Consortium. You can specify options indicating the coordinate system and number of coordinates for the values you are storing.
+
+The adapter extends ActiveRecord's migration syntax to
+support these spatial types. The following example creates five spatial
+columns in a table:
+
+```rb
+create_table :my_spatial_table do |t|
+  t.column :shape1, :geometry
+  t.geometry :shape2
+  t.line_string :path, srid: 3857
+  t.st_point :lonlat, geographic: true
+  t.st_point :lonlatheight, geographic: true, has_z: true
+end
+```
+
+The first column, "shape1", is created with type "geometry". This is a general
+"base class" for spatial types; the column declares that it can contain values
+of _any_ spatial type.
+
+The second column, "shape2", uses a shorthand syntax for the same type as the shape1 column.
+You can create a column either by invoking `column` or invoking the name of the type directly.
+
+The third column, "path", has a specific geometric type, `line_string`. It
+also specifies an SRID (spatial reference ID) that indicates which coordinate
+system it expects the data to be in. The column now has a "constraint" on it;
+it will accept only LineString data, and only data whose SRID is 3857.
+
+The fourth column, "lonlat", has the `st_point` type, and accepts only Point
+data. Furthermore, it declares the column as "geographic", which means it
+accepts longitude/latitude data, and performs calculations such as distances
+using a spheroidal domain.
+
+The fifth column, "lonlatheight", is a geographic (longitude/latitude) point
+that also includes a third "z" coordinate that can be used to store height
+information.
+
+The following are the data types understood by PostGIS and exposed by
+the adapter:
+
+- `:geometry` -- Any geometric type
+- `:st_point` -- Point data
+- `:line_string` -- LineString data
+- `:st_polygon` -- Polygon data
+- `:geometry_collection` -- Any collection type
+- `:multi_point` -- A collection of Points
+- `:multi_line_string` -- A collection of LineStrings
+- `:multi_polygon` -- A collection of Polygons
+
+Following are the options understood by the adapter:
+
+- `:geographic` -- If set to true, create a PostGIS geography column for
+  longitude/latitude data over a spheroidal domain; otherwise create a
+  geometry column in a flat coordinate system. Default is false. Also
+  implies :srid set to 4326.
+- `:srid` -- Set a SRID constraint for the column. Default is 4326 for a
+  geography column, or 0 for a geometry column. Note that PostGIS currently
+  (as of version 2.0) requires geography columns to have SRID 4326, so this
+  constraint is of limited use for geography columns.
+- `:has_z` -- Specify that objects in this column include a Z coordinate.
```

Comment:
`:has_m` too right?

Reply:
I had taken it out originially since the test for creating a polygon with an M coordinate failed due to an exception from Cockroach. Added it back though.